File: converted/utilities/qs/elide-qs.py
#!/usr/bin/env python3
"""
Python Integration Example for elide-qs

This demonstrates calling the TypeScript qs implementation from Python
using Elide's polyglot capabilities.

Benefits:
- One qs implementation shared across TypeScript and Python
- Consistent parse and stringify url query strings across services
- No Python custom implementation needed
- Perfect for API query parameters, URL generation, form data parsing
"""

# NOTE: Exact syntax depends on Elide's Python polyglot API (currently alpha)
# This is a CONCEPTUAL example - adjust when Python support is ready

# Assuming Elide provides something like:
# from elide import require
# qs_module = require('./elide-qs.ts')

print("=== Python Calling TypeScript qs ===\n")

# Example: Basic usage
# result = qs_module.default()
# print(f"Result: {result}")

print("Real-world use case:")
print("- Python service uses parse and stringify url query strings")
print("- Uses same TypeScript implementation as Node.js API")
print("- Guarantees consistent behavior across entire stack")
print("- Perfect for API query parameters, URL generation, form data parsing")
print()

print("Example: Microservices Architecture")
print("┌─────────────────────────────────────┐")
print("│   Elide qs (TypeScript)         │")
print("│   conversions/qs/elide-qs.ts│")
print("└─────────────────────────────────────┘")
print("         ↓                   ↓")
print("    ┌────────┐          ┌────────┐")
print("    │ Node.js│          │ Python │")
print("    │  API   │          │Pipeline│")
print("    └────────┘          └────────┘")
print("         ↓                   ↓")
print("    Same parse and stringify url query strings everywhere!")
print()

print("Problem Solved:")
print("Before: Python custom implementation + JavaScript = inconsistencies")
print("After: One Elide implementation = 100% consistent")
print()

print("When Elide Python API is ready, usage will be:")
print("  from elide import require")
print("  qs = require('./elide-qs.ts')")
print("  # Use it directly!")

File: converted/utilities/qs/elide-qs.rb
#!/usr/bin/env ruby

# Ruby Integration Example for elide-qs
#
# This demonstrates calling the TypeScript qs implementation from Ruby
# using Elide's polyglot capabilities.
#
# Benefits:
# - One qs implementation shared across TypeScript and Ruby
# - Consistent parse and stringify url query strings across microservices
# - Perfect for Rails, Sidekiq, background workers

puts "=== Ruby Calling TypeScript qs ===\n"

# NOTE: Exact syntax depends on Elide's Ruby polyglot API (currently alpha)
# This is a CONCEPTUAL example - adjust when Ruby support is ready

# Assuming Elide provides something like:
# qs_module = Elide.require('./elide-qs.ts')

# Example: Basic usage
# result = qs_module.default()
# puts "Result: #{result}"

puts "Real-world use case:"
puts "- Ruby worker uses parse and stringify url query strings"
puts "- Uses same TypeScript implementation as Node.js API"
puts "- Guarantees consistent behavior across services"
puts "- Perfect for API query parameters, URL generation, form data parsing"
puts ""

puts "Example: Microservices Architecture"
puts "┌─────────────────────────────────────┐"
puts "│   Elide qs (TypeScript)         │"
puts "│   conversions/qs/elide-qs.ts│"
puts "└─────────────────────────────────────┘"
puts "         ↓           ↓           ↓"
puts "    ┌────────┐  ┌────────┐  ┌────────┐"
puts "    │ Node.js│  │  Ruby  │  │ Rails  │"
puts "    │  API   │  │ Worker │  │  App   │"
puts "    └────────┘  └────────┘  └────────┘"
puts "         ↓           ↓           ↓"
puts "    Same parse and stringify url query strings everywhere!"
puts ""

puts "Problem Solved:"
puts "Before: Ruby custom + JavaScript = different behaviors"
puts "After: One Elide implementation = 100% consistent"
puts ""

puts "Rails Integration (when ready):"
puts "  # config/initializers/qs.rb"
puts "  QS = Elide.require('./elide-qs.ts')"
